### DISCUSSION TITLE
[HowTo] A script to switch Yakuake colors

### DISCUSSION THREAD

**Discussion by noyannus** [oskarsh/Yin-Yang#248] [Show and tell]
A first attempt to include Yakuake. I have not found a way to make changes stick in a running instance of Yakuake -- it must be restarted.

Copy it into a shell script and run it from the Custom plugin.

Also, how can one programmatically set the skin (title bar, tab bar), window?

```
PROFILE='Nordic'
for SESSID in $(qdbus org.kde.yakuake | grep /Sessions/) ; do
#     printf "\tsession: $SESSID\tprofile: '$PROFILE\'\n"

    # the profile is set via **konsole**
    qdbus org.kde.yakuake $SESSID org.kde.konsole.Session.setProfile "$PROFILE"
done

# TODO set default for new tabs in current window -- HOW??
    # does not affect currently open windows
    sed -i "s/DefaultProfile=.*/DefaultProfile=${PROFILE}\.profile/" ~/.config/yakuakerc
    # printf "profile writteprofile written to .config: \'${PROFILE}\'\n"
    # how to reload .config??

# TODO set skin -- HOW??
```

**Comment by l0drex** [comment]
Yakuake should already be supported with the Konsole plugin, same for the terminal in Dolphin. Tabs and window decorations seem to use their own theme, with only one Breeze theme by default. If you find a way, let me know! I would be happy to include a plugin for that.

**Comment by noyannus** [comment]
> Tabs and window decorations seem to use their own theme, with only one Breeze theme by default. If you find a way, let me know! I would be happy to include a plugin for that.

I've hammered away at that for another weekend, but eventually gave up. This is too entangled with plasma et al. for me.
